lib/dat-worker-pool/worker.rb

```diff
@@ -187,8 +189,6 @@ def initialize(worker_class, options = nil)
 
         def run(work_item)
           self.start
-          self.sleep
-          self.wakeup
           self.work(work_item)
           self.shutdown
         end
```

Comment:
@jcredding it sucks that we don't have more granularity on this.  The work method now implies 5 hooks for custom logic and you can't run them one at a time.  I'd love it if we could have it go: `make_unavailable`, `work`, `make_available`.  Is that possible?